lib/index.js

```diff
@@ -495,49 +495,57 @@ class AuthenticationPlugin {
 
     let sort;
     let query;
-    let isRequiredFieldsInBody = false;
-    const forbiddenFields = extractESMappingFields(storageMapping.users);
-    const forbiddenKeywords = [
-      'multi_match',
-    ]; // @todo Keep up to date this blacklist to prevent queries on forbiddenFields
+    let keywordWhitelist;
 
-    const sanitizeSearchBody = (key, value) => {
+    const sanitizeSearchBody = (key) => {
       if (key === 'username') {
-        isRequiredFieldsInBody = true;
         return { key: '_id' };
       }
-      else if (value === 'username') {
-        isRequiredFieldsInBody = true;
-        return { value: '_id' };
-      }
-      else if (forbiddenKeywords.includes(key)) {
+      else if (! keywordWhitelist.includes(key)) {
+        const forbiddenFields = extractESMappingFields(storageMapping.users);
+        if (forbiddenFields.includes(key)) {
+          throw new ForbiddenError(`Forbidden field "${key}". Only the "username" or "kuid" fields are sortable and only the first is also searchable.`);
```

Comment:
Why can't we search by `kuid` ?

Reply:
Because it's literally what we are looking for :sweat_smile: